### PR TITLE
Rename SA: build01 -> image-puller

### PR DESCRIPTION
**Pull request by hongkailiu**
This PR starts the work on https://issues.redhat.com/browse/DPTP-1500

I will clean those up at the end of this card on all clusters.

```
oc get sa build01
oc get clusterrolebinding image-puller-build01
```

Following up:
After merge, I will `make ci-secret-generator`.
Then rename the references of the generated file.

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
I don't think `apply` can change these immutable fields in the RBAC - can we add a new service account here to duplicate the values, then remove this when the BitWarden items are updated with that value?

**Review by petr-muller — APPROVED** [review]
/hold

This looks good to me, not sure if this needs some babysitting on merge, so holding.

File: core-services/ci-secret-generator/_config.yaml [hunk 1]
```diff
@@ -38,8 +38,8 @@
 - item_name: build_farm
   attachments:
   - cmd: |-
-      oc --context $(cluster) get secrets --namespace ci -o json | jq '.items[] | select(.type=="kubernetes.io/dockercfg") | select(.metadata.annotations["kubernetes.io/service-account.name"]=="build01") | .data[".dockercfg"]' --raw-output | base64 --decode | jq '.["image-registry.openshift-image-registry.svc:5000"].auth' --raw-output | tr -d '\n'
-    name: build01_$(cluster)_reg_auth_value.txt
+      oc --context $(cluster) get secrets --namespace ci -o json | jq '.items[] | select(.type=="kubernetes.io/dockercfg") | select(.metadata.annotations["kubernetes.io/service-account.name"]=="image-puller") | .data[".dockercfg"]' --raw-output | base64 --decode | jq '.["image-registry.openshift-image-registry.svc:5000"].auth' --raw-output | tr -d '\n'
+    name: image-puller_$(cluster)_reg_auth_value.txt
```
Comment (stevekuznetsov) [inline core-services/ci-secret-generator/_config.yaml:42]: While we're at it ... maybe `token.image-puller.$(cluster).txt` to match the other values we have in here?

Comment (hongkailiu) [reply core-services/ci-secret-generator/_config.yaml]: Updated.
I do not see other "token" in this file though.

**Review by petr-muller — APPROVED** [review]

**Comment by hongkailiu** [comment]
Thanks for the review, @petr-muller 
I will cancel the hold when I resume the work on the story.
(currently i am occupied by migrating postsubmits)

**Pull request merged**